### ISSUE TITLE
[HOLD] [$500] [CRITICAL] Delegated Splits - Allow selecting a payer from the splits page

### ISSUE THREAD

**Issue by youssef-lr** [Expensify/App#40379]
Implement this section of the [Uneven Split Design Doc](https://docs.google.com/document/d/1wOB61kuktAKbvMsg8dAm9FaO_bhZepXe8agQtvvpgk4/edit#heading=h.mnjguil21u0a) (the section is copied below if you can't access the document)

## Context

Currently when creating a split, the current user is automatically set as the payer of the split and the person that needs to be paid back by the other participants.

## Problem

We currently don't allow selecting a different payer of the split. There are real world use cases where we'd want to create a split on behalf of someone else. e.g.

>You are heavily entrenched in the NewDot ecosystem and would like to utilize split with your social group. However, they are resistant to trying a new app. With Group Chat and delegated splits, you are able to create splits on their behalf. You are able to continue leveraging NewDot, your social circle can enjoy the benefits of NewDot in a hands-off environment, and Expensify reaps the benefits of a highly viral, member-driven strategy for gaining new customers. 

## Solution

<img width="595" alt="Screenshot 2024-04-17 at 21 52 54" src="https://github.com/Expensify/App/assets/9680864/b94a656d-0b7c-4c44-889a-f928cce2623d">

Enhance the split page by allowing the selection of a different payer by creating a payer selector page, which should display all of the splits participants and allow selecting one of them. 

## Implementation details

1. We'll begin by adding a `MenuItem` in the `MoneyTemporaryForRefactorRequestConfirmationList` that will allow us to navigate to the new page:

```
<MenuItem
    label={translate('moneyRequestConfirmationList.paidBy')}
    description={personalDetailsOfPayer.login}
    title={personalDetailsOfPayee.displayName}
    icon={personalDetailsOfPayee.icons}
    onPress={() => }}
    shouldShowRightIcon
    titleWithTooltips={payeeTooltipDetails}
/>
```

_Note: in the current code it's called a `payee` instead of payer. That's because the code is meant for all types of IOUs, we can keep it as `payee`, but in the context of splits the `payee` is the same as the `payer` (as they paid for the split bill)._

The `icon` prop passed to the menu item is what allows us to display their avatar, so we should make use of `ReportUtils.getIcons` to retrieve them for the selected payer accountID by passing their `personalDetails` to it, which is already available in the confirmation page component.

2. Create a new IOU request step page, called `IOURequestStepSplitPayer` This will work very similar to how we select a task assignee (we can copy most of the code from that [page](https://github.com/Expensify/App/blob/b1b8b4c6aa42e2af89b442ea75ac6a31530cdb10/src/pages/tasks/TaskAssigneeSelectorModal.tsx) and adjust it), the navigation to this page will be triggered from the `onPress` handler in the `MenuItem` above.

3. The navigation config of this new page should be consistent with other steps like amount, merchant, etc. It should have this route: `create/split/confirmation/<transactionID>/<reportActionID>/payer`

4. The page will subscribe to the `TRANSACTION_DRAFT` Onyx key which will allow it to retrieve the split participants which we can then display similar to how we display a report’s participants, except we’ll also allow selecting one user from the list. The selected payer should have a green check mark next to it. 

5. Once a user is selected, it will call a new IOU action which will set the payer accountID in the transaction draft and then navigate the user back to the split page:

```
function setSplitPayer(transactionID: string, payerAccountID: number) {
    Onyx.merge(`${ONYXKEYS.COLLECTION.TRANSACTION_DRAFT}${transactionID}`, {splitPayerAccountIDs: [payerAccountID]});
}
```
_Note: we're planning to support multiple payers in the future. For now the `splitPayerAccountIDs` should hold a single accountID._

7. Finally, we need to send the array of `splitPayerAccoutIDs` to the backend [here](https://github.com/Expensify/App/blob/b1b8b4c6aa42e2af89b442ea75ac6a31530cdb10/src/libs/actions/IOU.ts#L2945) which should be present in the Onyx transaction. Passed to `splitBill` from [here](https://github.com/Expensify/App/blob/b1b8b4c6aa42e2af89b442ea75ac6a31530cdb10/src/pages/iou/request/step/IOURequestStepConfirmation.tsx#L296).

<details><summary>Issue Owner</summary>Current Issue Owner: @</details>
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01da88a2530aad1346</li>
        <li>Upwork Job ID: 1787475123627106304</li>
        <li>Last Price Increase: 2024-05-06</li>
    </ul>
</details>

**Comment by nkdengineer** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Allow selecting a payer from the splits page
### What is the root cause of that problem?
This is a new feature
### What changes do you think we should make in order to solve the problem?
Since the implementation is detail enough and no need to add more solution. I'd love to work on this issue.


### What alternative solutions did you explore? (Optional)
NA

**Comment by nkdengineer** [comment]
@youssef-lr Do I need to update more detail in my proposal? I see that the implementation here https://github.com/Expensify/App/issues/40379#issue-2249250830 is detail enough.

**Comment by youssef-lr** [comment]
@nkdengineer Not really! If you can give top priority, I'm ready to assign you!

**Comment by nkdengineer** [comment]
@youssef-lr   Sure, I can priority this.

**Comment by youssef-lr** [comment]
Awesome, thanks!

**Comment by kevinksullivan** [comment]
Hi @nkdengineer thanks for taking this on! Full disclosure, this PR is associated with a deadline, so we need to get this done as quickly as possible. Like @youssef-lr said, this should be prioritized above any other work you are doing for us.

Also, please provide a daily update along with an ETA of when you expect this PR to be merged. As soon as you need a review or are blocked on moving this forward, post here and DM me on slack. Thank you!

**Comment by rafecolton** [comment]
Taking over management of this issue from @youssef-lr

**Comment by nkdengineer** [comment]
> Hi @nkdengineer thanks for taking this on! Full disclosure, this PR is associated with a deadline, so we need to get this done as quickly as possible. Like @youssef-lr said, this should be prioritized above any other work you are doing for us.

Sure, I will priority this issue.

> Also, please provide a daily update along with an ETA of when you expect this PR to be merged. As soon as you need a review or are blocked on moving this forward, post here and DM me on slack. Thank you!

I'm working on the PR, will open it in a few hours.

**Comment by rafecolton** [comment]
Thanks @nkdengineer! I'll be on pretty early tomorrow, so I'm looking forward to seeing your progress! Is your Slack handle `@NKDengineer`?

**Comment by nkdengineer** [comment]
> Thanks @nkdengineer! I'll be on pretty early tomorrow, so I'm looking forward to seeing your progress! Is your Slack handle @NKDengineer?

@rafecolton Yes, this is my slack.

**Comment by nkdengineer** [comment]
@dukenv0307  The PR is [here](https://github.com/Expensify/App/pull/40388).

**Comment by nkdengineer** [comment]
@rafecolton @joekaufmanexpensify Please add `Waiting for copy` label, we need to confirm a new translation key

- English: `Split amounts`
- Spanish: `Dividir importes`

**Comment by rafecolton** [comment]
Added the label and asked in Slack as well [here](https://expensify.slack.com/archives/C21FRDWCV/p1713464994024569) (internal only)

**Comment by rafecolton** [comment]
@nkdengineer the translation you're looking for is `Importes a dividir`

**Comment by joekaufmanexpensify** [comment]
PR's in active review

**Comment by joekaufmanexpensify** [comment]
Moving to weekly, since PR is merged!

**Comment by joekaufmanexpensify** [comment]
@dukenv0307 could you please handle BZ so we can prep for payment?

**Comment by joekaufmanexpensify** [comment]
Payment here is going to be: 

- @nkdengineer - $500 for PR review via upwork
- @dukenv0307 - $500 for C+ review via upwork

**Comment by joekaufmanexpensify** [comment]
offer sent to both of you!

**Comment by joekaufmanexpensify** [comment]
@dukenv0307 could you clarify a bit about what you mean by enabling that feature?

**Comment by joekaufmanexpensify** [comment]
Also @dukenv0307 could you please complete the BZ checklist so we can issue payment here? We're otherwise all set to do that.

**Comment by youssef-lr** [comment]
@dukenv0307 @joekaufmanexpensify there's still some to finish in an ongoing draft PR before we can enable that menu item to select a different payer. As well as backend changes being deployed which isn't done yet.

**Comment by joekaufmanexpensify** [comment]
@youssef-lr Got it. Are we going to handle those follow up PRs in this issue? If so, perhaps we should issue payment when the total issue is done, unless none of the remaining PRs are frontend and won't be worked on by @dukenv0307 or @nkdengineer

**Comment by youssef-lr** [comment]
PR has been worked by @rafecolton and myself @joekaufmanexpensify, I think @dukenv0307 can review the remaining PR but we'll have to adjust the price for him, as the changes in this new PR were not included in the scope of the first one.

**Comment by joekaufmanexpensify** [comment]
@youssef-lr mind linking the PR here?

**Comment by joekaufmanexpensify** [comment]
@youssef-lr or @rafecolton when you have a sec, mind linking this additional PR here so we can get @dukenv0307 going on reviewing it (assuming he should review it, and is not already).

**Comment by rafecolton** [comment]
The PR is https://github.com/Expensify/App/pull/41375 but it's still in draft and not ready for review. We also need to finish https://github.com/Expensify/Web-Expensify/pull/41775 and https://github.com/Expensify/Auth/pull/10631 before this feature is "done"

**Comment by joekaufmanexpensify** [comment]
Cool, sounds good! For @dukenv0307's payment, I will hold off for now if they're going to review that additional app PR.

**Comment by joekaufmanexpensify** [comment]
@rafecolton is there an ETA for having the follow on App PR in review?

**Comment by joekaufmanexpensify** [comment]
Sounds good! @youssef-lr mind weighing in on an ETA for the app PR getting to review when you have a sec?

**Comment by youssef-lr** [comment]
Should be ready sometime early next week!

**Comment by joekaufmanexpensify** [comment]
Great. TY!

**Comment by joekaufmanexpensify** [comment]
App PR still in progress

**Comment by joekaufmanexpensify** [comment]
Same

**Comment by joekaufmanexpensify** [comment]
PR still in progress

**Comment by youssef-lr** [comment]
Holding this until we fix these issues and I'll continue working on it.

https://github.com/Expensify/Expensify/issues/323235
https://github.com/Expensify/Expensify/issues/395703
https://github.com/Expensify/App/issues/42021
https://github.com/Expensify/App/issues/30140

**Comment by youssef-lr** [comment]
@joekaufmanexpensify I think we can issue payments here, @nkdengineer and @dukenv0307 have worked on the original scope of the issue. This might take a while before I continue working on this as we need to refactor things a bit before continuing with the implementation https://expensify.slack.com/archives/C05RECHFBEW/p1716809169081189

**Comment by joekaufmanexpensify** [comment]
Payment summary is [here](https://github.com/Expensify/App/issues/40379#issuecomment-2096030983).

**Comment by joekaufmanexpensify** [comment]
@nkdengineer $500 sent and contract ended!

**Comment by joekaufmanexpensify** [comment]
Upwork job closed.

**Comment by joekaufmanexpensify** [comment]
internal portions are held

**Comment by youssef-lr** [comment]
We've paused working on this based on https://expensify.slack.com/archives/CC7NECV4L/p1717525527830119

**Comment by joekaufmanexpensify** [comment]
Sounds good!

**Comment by joekaufmanexpensify** [comment]
Still paused

**Comment by rafecolton** [comment]
Still paused

**Comment by joekaufmanexpensify** [comment]
Same

**Comment by joekaufmanexpensify** [comment]
Same.

**Comment by joekaufmanexpensify** [comment]
@youssef-lr is there anything left here, or are we good to close this one? The #vip-split project got closed in GH, so trying to determine whether we should move this to another project, or just close.
